### PR TITLE
How to Implement K-Fold Cross-Validation with Scikit-Learn

### LINKED ISSUE #5223: [Machine Learning] How to Implement K fold Cross-Validation in Scikit-Learn
### Topic Suggestion
[Machine Learning] How to Implement K fold Cross-Validation in Scikit-Learn

#### Proposed title of the article
How to Implement K fold Cross-Validation in Scikit-Learn

#### Proposed article introduction
In machine learning, the most common approach is to split a dataset into the training set and the validation set. The splitting ratio could be 70:30 or 80:20. The problem with this approach is we are not sure if a good validation accuracy means the model is really good. What if the portion of the data used as validation luckily gave us a good result? If we had used another portion of the dataset as a validation set, would our model still give us a good accuracy? These are some of the things we are not sure of. K fold cross-validation provides a better way of validating our model performance. It evaluates the model using different portions of the dataset as the validation set in various iterations.

The K fold cross-validation technique is helpful when you are training your model on a small dataset. If your dataset is large (maybe over 100,000 samples), you may not have to use K fold cross-validation. The reason is you will have enough data records in your validation set to evaluate the machine learning model. Also, using the K fold cross-validation on a large dataset will take a lot of time.

#### Key takeaways 
What you will learn:
1.	How to use K fold cross-validation to measure the performance of the model.
2.	How to use the decision tree algorithm in Sci-kit Learn.
3.	How to visualize your training accuracy, training recall, training precision and training f1-score on each fold of the dataset.
4.	How to visualize your validation accuracy, validation recall, validation precision and validation f1-score on each fold of the dataset.
5. How to reduce overfitting in a decision trees model.

#### Article quality
This article is in-depth. It will teach people how to visualize a machine learning model performance. They will be able to tell if the model is overfitting or underfitting. I realized that most of the models trained using decision trees overfit. If you can visualize the training and validation performance of the model, you’ll be able to prevent this by adjusting the hyperparameters of your decision tree model.

### PR DESCRIPTION
**Pull request by Inyrkz**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5223 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by lalith1403** [comment]
👋 Good afternoon
Many sentences in this article appear to be taken from third-party sources. We can't publish any content that is not original or adequately cited/referenced. There are many free online tools where you can check this (such as quetext.com, and if you're uncertain about the definitions around plagiarism, this is an excellent resource to reference.
We're looking for unique/original content.

[How to Implement K-Fold Cross-Validation with Scikit-Learn #5592.pdf](https://github.com/section-engineering-education/engineering-education/files/7721795/How.to.Implement.K-Fold.Cross-Validation.with.Scikit-Learn.5592.pdf)
